kyo-core/shared/src/test/scala/kyoTest/KyoTestZIO.scala

```diff
@@ -0,0 +1,78 @@
+package kyoTest
+
+import kyo.*
+import kyo.KyoApp.Effects
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.concurrent.duration.Duration
+import scala.language.postfixOps
+import zio.ZIO
+import zio.ZLayer
+import zio.internal.stacktracer.SourceLocation
+import zio.test.*
+
+abstract class KyoTestZIO[S] extends ZIOSpecAbstract:
+    final val environmentTag: zio.EnvironmentTag[Any] = zio.EnvironmentTag[Any]
+
+    private val trace: TestAnnotation[List[SourceLocation]] =
+        TestAnnotation("trace", List.empty, _ ++ _)
+
+    override type Environment = Any
+    override def bootstrap: ZLayer[Any, Any, Any] = ZLayer.empty
+
+    def spec: Spec[Any, Any]
+    def timeout: Duration = Duration.Inf
+    def run[In](v: In < S)(using ExecutionContext, Flat[In < S]): Future[In]
+
+    def test[In <: TestResult](label: String)(assertion: In < S)(
+        using
+        fl: Flat[In < S],
+        sl: SourceLocation,
+        tr: zio.Trace
+    ): Spec[Any, Throwable] =
+        Spec.labeled(
+            label,
+            Spec
+                .test(ZTest(label, ZIO.fromFuture { implicit ec => run(assertion) }), TestAnnotationMap.empty)
+                .annotate(trace, sl :: Nil)
+        )
+
+    def suite[In](label: String)(specs: In*)(using
+        suiteConstructor: SuiteConstructor[In],
+        sourceLocation: SourceLocation,
+        trace: zio.Trace
+    ): Spec[suiteConstructor.OutEnvironment, suiteConstructor.OutError] =
+        zio.test.suite(label)(specs*)
+
+end KyoTestZIO
+
+abstract class KyoTestDefault extends KyoTestZIO[KyoApp.Effects]:
+    final override def run[In](v: In < KyoApp.Effects)(using ExecutionContext, Flat[In < KyoApp.Effects]): Future[In] =
+        IOs.run(KyoApp.runFiber(timeout)(IOs(v)).toFuture).map(_.get)
+
+object ExampleSpec extends KyoTestDefault:
+    def spec: Spec[Any, Throwable] =
+        suite("suite!")(
+            test("pure") {
+                assertTrue(true)
+            },
+            test("IOs Succeed") {
+                IOs(assertTrue(true))
+            },
+            test("IOs Fail") {
+                for
+                    _ <- IOs.fail("ERROR!")
+                yield assertTrue(true)
+            } @@ TestAspect.failing,
+            test("Aborts!") {
+                for
+                    _ <- Aborts[Throwable].fail(new RuntimeException("Aborts!"))
+                yield assertTrue(true)
+            } @@ TestAspect.failing,
+            test("Fibers.sleep") {
```

Comment:
```
[info]   - Fibers.sleep
[info] Timeout of 0 ns exceeded.
[info] 4 tests passed. 1 tests failed. 0 tests ignored.
[info]   - suite! - Fibers.sleep
[info] Timeout of 0 ns exceeded.
[info] Executed in 288 ms
[info] timestamp=2024-04-24T06:22:45.194897Z level=ERROR thread=#zio-fiber-1 message="" cause="Exception in thread "zio-fiber-4" java.lang.RuntimeException: null
[info]  at zio.test.ZIOSpecAbstract.runSpec$$anonfun$1$$anonfun$2$$anonfun$1$$anonfun$3$$anonfun$2$$anonfun$1(ZIOSpecAbstract.scala:79)
[info]  at zio.ZIO$.fail$$anonfun$1(ZIO.scala:3151)
[info]  at zio.ZIO$.failCause$$anonfun$1$$anonfun$1$$anonfun$1(ZIO.scala:3160)"
```

Can we improve this error message? ie should the exception have a message?

Reply:
It looks like there are two causes for the bad error messages. When timing out fibers directly (from Kyo code), the `NoStackTrace` exception is throwing with a message of `null`.  This can be fixed via adding a message here:

```
    case object Interrupted
        extends RuntimeException("Fiber Interrupted")
        with NoStackTrace

    private[kyo] val interrupted = IOs.fail(Interrupted)
```

However, this still does not solve the `cause="Exception in thread "zio-fiber-4" java.lang.RuntimeException: null`. Will need to dig in further.

Reply:
Fixed with latest commit:

```
[info]   - Fibers.sleep
[info] Timeout of 0 ns exceeded.
[info] 4 tests passed. 1 tests failed. 0 tests ignored.
[info]   - suite! - Fibers.sleep
[info] Timeout of 0 ns exceeded.
[info] Executed in 248 ms
[info] timestamp=2024-04-24T19:33:53.526752Z level=ERROR thread=#zio-fiber-1761416989 message="" cause="Exception in thread "zio-fiber-1308002466" java.lang.RuntimeException: Tests failed.
[info]  at zio.test.ZIOSpecAbstract.runSpec$$anonfun$1$$anonfun$2$$anonfun$1$$anonfun$3$$anonfun$2$$anonfun$1(ZIOSpecAbstract.scala:80)
[info]  at zio.ZIO$.fail$$anonfun$1(ZIO.scala:3154)
[info]  at zio.ZIO$.failCause$$anonfun$1$$anonfun$1$$anonfun$1(ZIO.scala:3163)"

```

I think this is fixed by switching to ZIO 2.1 which has improved ZIO.fromFuture